job_bundles/Using_Files_In_Jobs.md

```diff
@@ -0,0 +1,1161 @@
+
+```bash
+# Change the value of WSALL_ID to the identifier of the WSAll storage profile
+WSALL_ID=sp-00112233445566778899aabbccddeeff
+
+sudo mkdir -p /shared/common /shared/projects/project1 /shared/projects/project2
+sudo chown -R cloudshell-user:cloudshell-user /shared
+
+for d in /shared/common /shared/projects/project1 /shared/projects/project2; do
+  echo "File contents for $d" > ${d}/file.txt
+done
+```
+
+Next, add an [asset references file](README#elements---asset-references) to the job bundle that includes all of the files that you created as inputs for the job:
+
+```yaml
+cat > ${HOME}/job_attachments_devguide/asset_references.yaml << EOF
+assetReferences:
+  inputs:
+    filenames:
+    - /shared/common/file.txt
+    directories:
+    - /shared/projects/project1
+    - /shared/projects/project2
+EOF
+```
+
+Then, configure the Deadline Cloud CLI to submit jobs with the `WSAll` storage profile, and submit the job bundle:
+
+```bash
+# Change the value of FARM_ID to your farm's identifier
+FARM_ID=farm-00112233445566778899aabbccddeeff
+# Change the value of QUEUE1_ID to queue Q1's identifier
+QUEUE1_ID=queue-00112233445566778899aabbccddeeff
+# Change the value of WSALL_ID to the identifier of the WSAll storage profile
+WSALL_ID=sp-00112233445566778899aabbccddeeff
+
+deadline config set settings.storage_profile_id $WSALL_ID
+
+deadline bundle submit --farm-id $FARM_ID --queue-id $QUEUE1_ID job_attachments_devguide/
+```
+
+Two files are uploaded to Amazon S3 when the job is submitted. You can download the manifest objects for the job from Amazon S3
+to see which files were uploaded:
+
+```bash
+for manifest in $( \
+  aws deadline get-job --farm-id $FARM_ID --queue-id $QUEUE1_ID --job-id $JOB_ID \
+    --query 'attachments.manifests[].inputManifestPath' \
+    | jq -r '.[]'
+); do
+  echo "Manifest object: $manifest"
+  aws s3 cp --quiet s3://$Q1_S3_BUCKET/DeadlineCloud/Manifests/$manifest /dev/stdout | jq .
+done
+```
+
+In this case, there is a single manifest file with contents:
+
+```json
+{
+    "hashAlg": "xxh128",
+    "manifestVersion": "2023-03-03",
+    "paths": [
+        {
+            "hash": "87cb19095dd5d78fcaf56384ef0e6241",
+            "mtime": 1721147454416085,
+            "path": "home/cloudshell-user/job_attachments_devguide/script.sh",
+            "size": 39
+        },
+        {
+            "hash": "af5a605a3a4e86ce7be7ac5237b51b79",
+            "mtime": 1721163773582362,
+            "path": "shared/projects/project2/file.txt",
+            "size": 44
+        }
+    ],
+    "totalSize": 83
+}
+```
+
+Look at the result from the GetJob API for this manifest and you will see that the `rootPath` is "/" for this manifest:
+
+```bash
+aws deadline get-job --farm-id $FARM_ID --queue-id $QUEUE1_ID --job-id $JOB_ID --query 'attachments.manifests[*]'
+```
+
+The root path of a set of input files is always the longest common subpath of those files. If your job was submitted from, say,
```

Comment:
Nit: I'm preeeetty sure the root path will also be split based on any `LOCAL` storage profile locations, and then all of the paths that fall outside of all local locations will be grouped under a common path

Reply:
Discussed offline.